packages/core/helper-plugin/lib/src/utils/getAPIInnerErrors/index.js

```diff
@@ -1,16 +1,30 @@
+function getErrorPayload(error) {
+  const response = error.response.data;
+
+  if (['YupValidationError', 'ValidationError'].includes(response.error.name)) {
+    return response.error.details.errors;
+  }
+
+  return [response.error];
+}
+
 export default function getAPIInnerErrors(error, { getTrad }) {
-  const errorPayload = error.response.data.error.details.errors;
-  const validationErrors = errorPayload.reduce((acc, err) => {
-    acc[err.path.join('.')] = {
-      id: getTrad(`apiError.${err.message}`),
-      defaultMessage: err.message,
-      values: {
-        field: err.path[err.path.length - 1],
-      },
+  return getErrorPayload(error).reduce((acc, err, index) => {
+    const path = err?.path ? err?.path.join('.') : `${err.name}.${index}`;
```

Comment:
why are you still checking the presence of the path in err, you already did it in the condition, I think it is better to write
```
return getErrorPayload(error).reduce((acc, err, index) => {
    const path = err?.path ? err.path.join('.') : 
```
wdyt?

Reply:
I've refactored this bit and split `ValidationError` more from `ApplicationError`. I think this will help us in the future to better handle API errors in FE.